tests/backends/tensorflow/test_gradient.py

```diff
@@ -133,7 +133,7 @@ def test_Displacement_fock_probabilities_jacobian_2_modes():
 
 
 def test_Squeezing_mean_photon_number_gradient_1_mode():
-    r = tf.Variable(0.1)
+    r = tf.Variable(0.1, dtype=np.double)
```

Comment:
I got an error that it needed double precision instead of float. Maybe it is due to something similar like what we had last week or so with the scipy precision error. I'll check again without it.

Reply:
`tensorflow.python.framework.errors_impl.InvalidArgumentError: cannot compute AddN as input #1(zero-based) was expected to be a double tensor but is a float tensor [Op:AddN]`
After deleting that.

Reply:
Interesting. It should be able to perform the simulation either way. Where does it occur in the code?

Reply:
This is the full message: 

```def test_Squeezing_mean_photon_number_gradient_1_mode():
        r = tf.Variable(0.1)

        simulator = pq.TensorflowPureFockSimulator(d=1, config=pq.Config(cutoff=8))

        with tf.GradientTape() as tape:
            with pq.Program() as program:
                pq.Q() | pq.Vacuum()

                pq.Q(0) | pq.Squeezing(r=r)

            state = simulator.execute(program).state

            mean = state.mean_photon_number()

>       gradient = tape.gradient(mean, [r])

tests/backends/tensorflow/test_gradient.py:150:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
.venv/lib/python3.10/site-packages/tensorflow/python/eager/backprop.py:1112: in gradient
    flat_grad = imperative_grad.imperative_grad(
.venv/lib/python3.10/site-packages/tensorflow/python/eager/imperative_grad.py:67: in imperative_grad
    return pywrap_tfe.TFE_Py_TapeGradient(
.venv/lib/python3.10/site-packages/tensorflow/python/eager/backprop.py:647: in _aggregate_grads
    return gen_math_ops.add_n(gradients)
.venv/lib/python3.10/site-packages/tensorflow/python/ops/gen_math_ops.py:398: in add_n
    _ops.raise_from_not_ok_status(e, name)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

e = _NotOkStatusException(), name = None

    def raise_from_not_ok_status(e, name):
      e.message += (" name: " + name if name is not None else "")
>     raise core._status_to_exception(e) from None  # pylint: disable=protected-access
E     tensorflow.python.framework.errors_impl.InvalidArgumentError: cannot compute AddN as input #1(zero-based) was expected to be a double tensor but is a float tensor [Op:AddN]```